### PR TITLE
feat:adds export for Log Bucket

### PR DESCRIPTION
**Pull request by g-awmalik**

### CODE REVIEW

**Comment by g-awmalik** [comment]
@bharathkkb - can you sanity check this please?

File: functions/go/export-terraform/terraformgenerator/templates/log-export.tf [hunk 1]
```diff
@@ -45,4 +45,14 @@ module "{{ .GetResourceName }}-destination" {
     is_locked             = {{ .GetBool "spec" "retentionPolicy" "isLocked" }}
   }{{end}}
 }
+{{end}}{{with $logsink.References.LoggingLogBucket }}
+module "{{ .GetResourceName }}-destination" {
+  source  = "terraform-google-modules/log-export/google//modules/logbucket"
+  version = "~> 7.3.0"
```
Comment (bharathkkb) [inline functions/go/export-terraform/terraformgenerator/templates/log-export.tf:51]: Will only land in the next release
```suggestion
  version = "~> 7.4.0"
```

File: functions/go/export-terraform/terraformgenerator/templates/log-export.tf [hunk 2]
```diff
@@ -45,4 +45,14 @@ module "{{ .GetResourceName }}-destination" {
...
+
+  project_id                  = module.{{ .Parent.GetResourceName }}.project_id
+  log_bucket_name             = "{{ .GetResourceName }}"
```
Comment (bharathkkb) [inline functions/go/export-terraform/terraformgenerator/templates/log-export.tf:54]: Shouldn't this be name?
https://github.com/terraform-google-modules/terraform-google-log-export/blob/eb777cae1f7a7ef7df4f7f9651797b0213b3412f/modules/logbucket/variables.tf#L22

File: functions/go/export-terraform/terraformgenerator/templates/log-export.tf [hunk 3]
```diff
@@ -45,4 +45,14 @@ module "{{ .GetResourceName }}-destination" {
...
+  location                    = "{{.}}"{{end}}{{ with .GetStringFromObject "spec" "location" }}
+  retention_days              = "{{.}}"{{end}}{{ if .GetInt "spec" "retentionDays"}}
+}
```
Comment (bharathkkb) [inline functions/go/export-terraform/terraformgenerator/templates/log-export.tf:57]: We should also expose https://github.com/GoogleCloudPlatform/blueprints/blob/aa4a37d846c816de4649fa555c327f9ede89cb26/catalog/log-export/folder/logbucket-export/export.yaml#L58 in TF for parity above with the storage

**Review by bharathkkb — COMMENTED** [review]
@g-awmalik some comments below and lets also add a test to validate expected TF.

File: functions/go/export-terraform/terraformgenerator/templates/log-export.tf (hunk 1 above)
Comment (g-awmalik) [reply functions/go/export-terraform/terraformgenerator/templates/log-export.tf:51]: ack

File: functions/go/export-terraform/terraformgenerator/templates/log-export.tf (hunk 2 above)
Comment (g-awmalik) [reply functions/go/export-terraform/terraformgenerator/templates/log-export.tf:54]: Yes, you're right.

File: functions/go/export-terraform/terraformgenerator/templates/log-export.tf (hunk 3 above)
Comment (g-awmalik) [reply functions/go/export-terraform/terraformgenerator/templates/log-export.tf:57]: I wanted to but it's not exposed in the [TF resource](https://registry.terraform.io/providers/hashicorp/google/latest/docs/resources/logging_project_bucket_config) for some reason. I'll open up an issue in the provider repo but for now we'll have to let this go.

Comment (bharathkkb) [reply functions/go/export-terraform/terraformgenerator/templates/log-export.tf:57]: @g-awmalik sgtm

Comment (g-awmalik) [reply functions/go/export-terraform/terraformgenerator/templates/log-export.tf:57]: https://github.com/hashicorp/terraform-provider-google/issues/11798

**Comment by g-awmalik** [comment]
@mengqiy - the upsert-resource-simple example is erroring out with `kpt@main`. I've tried running it with the latest version of kpt i.e. 1.0.0-beta.16 is goes through fine. 

Seems like there's a feature (or bug) that doesn't allow two resources with the same GVK loaded in the list which is what the upsert example is trying to do.

Is there a reason we're using the `main` version for the CI? Can we switch that to a more stable (latest) version?

**Comment by mengqiy** [comment]
> Is there a reason we're using the main version for the CI? Can we switch that to a more stable (latest) version?

We use the kpt built from HEAD of `main` on purpose, since if a new kpt feature can break a function, we want to be aware of it before our users. Using the latest kpt can help us catch this signal earlier.

**Comment by g-awmalik** [comment]
Thanks for the update.

**Comment by g-awmalik** [comment]
@mengqiy - can we merge this since the tests for this function are all passing? This required for a feature we're working on for Customer Onboarding and we need to start testing soon.

**Comment by mengqiy** [comment]
@g-awmalik The e2e CI has been fixed on HEAD. A rebase on top of HEAD should make the tests passing.

File: functions/go/export-terraform/terraformgenerator/resources.go [hunk 4]
```diff
@@ -206,6 +218,8 @@ func (resource *terraformResource) attachReferences() error {
 		{kind: "ComputeRouter", path: []string{"spec", "routerRef", "name"}},
 		{kind: "ComputeAddress", customRetriever: singleComputeAddressRetriever([]string{"spec", "natIps"})},
 		{kind: "ComputeAddress", customRetriever: singleComputeAddressRetriever([]string{"spec", "reservedPeeringRanges"})},
+		//todo:awmalik@ - remove customerRetriver when this issue is addressed: https://github.com/GoogleCloudPlatform/k8s-config-connector/issues/665
```
Comment (mengqiy) [inline functions/go/export-terraform/terraformgenerator/resources.go:221]: nit: We should use `TODO` by convention.

**Review by mengqiy — COMMENTED** [review]
Please update the usage doc to reflect the change.
PR LGTM otherwise.

File: functions/go/export-terraform/terraformgenerator/resources.go (hunk 4 above)
Comment (g-awmalik) [reply functions/go/export-terraform/terraformgenerator/resources.go]: Ack

**Comment by g-awmalik** [comment]
@mengqiy - ping.

**Comment by mengqiy** [comment]
@g-awmalik Has you updated the usage doc (https://github.com/GoogleContainerTools/kpt-functions-catalog/blob/master/functions/go/export-terraform/README.md) as I requested in https://github.com/GoogleContainerTools/kpt-functions-catalog/pull/873#pullrequestreview-995488436.
It seems `LoggingLogBucket` is something new that's not reflected in the usage doc.

**Review by mengqiy — APPROVED** [review]

**Comment by g-awmalik** [comment]
Thanks for the approval @mengqiy. I don't think I can merge though, will need your help with that.

**Pull request merged**